contracts/BSCValidatorSet.sol

```diff
@@ -393,13 +398,15 @@ contract BSCValidatorSet is IBSCValidatorSet, System, IParamSubscriber, IApplica
       _maxNumOfWorkingCandidates, validators.length-_numOfCabinets+_maxNumOfWorkingCandidates);
     }
     address[] memory miningValidators = new address[](_numOfCabinets);
+    bytes[] memory miningVoteAddrs = new bytes[](_numOfCabinets);
     for (uint i=0;i<_numOfCabinets;i++) {
       miningValidators[i] = validators[i];
+      miningVoteAddrs[i] = voteAddrs[i];
```

Comment:
You can not simple assign `voteAddrs[i]` here, because validators have been shuffled, while voteAddrs not, they are not match already.

Reply:
Fixed